test/e2e/storage/csistoragecapacity.go

```diff
@@ -0,0 +1,209 @@
+		framework.ExpectNoError(err)
+
+		// added for a watch
+		_, err = cscClient.Create(context.TODO(), csc3, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("getting")
+		gottenCSC, err := cscClient.Get(context.TODO(), csc.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenCSC.UID, createdCSC.UID)
+
+		ginkgo.By("listing")
+		cscs, err := cscClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "test=" + f.UniqueName})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(len(cscs.Items), 3, "filtered list should have 3 items")
+
+		ginkgo.By("patching")
+		patchedCSC, err := cscClient.Patch(context.TODO(), createdCSC.Name, types.MergePatchType, []byte(`{"metadata":{"annotations":{"patched":"true"}}}`), metav1.PatchOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(patchedCSC.Annotations["patched"], "true", "patched object should have the applied annotation")
+
+		ginkgo.By("updating")
+		csrToUpdate := patchedCSC.DeepCopy()
+		csrToUpdate.Annotations["updated"] = "true"
+		updatedCSC, err := cscClient.Update(context.TODO(), csrToUpdate, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(updatedCSC.Annotations["updated"], "true", "updated object should have the applied annotation")
+
+		framework.Logf("waiting for watch events with expected annotations")
+		for sawAdded, sawPatched, sawUpdated := false, false, false; !sawAdded && !sawPatched && !sawUpdated; {
+			select {
+			case evt, ok := <-cscWatch.ResultChan():
+				framework.ExpectEqual(ok, true, "watch channel should not close")
+				if evt.Type == watch.Modified {
+					watchedCSC, isCSC := evt.Object.(*storagev1.CSIStorageCapacity)
+					framework.ExpectEqual(isCSC, true, fmt.Sprintf("expected CSC, got %T", evt.Object))
+					if watchedCSC.Annotations["patched"] == "true" {
+						framework.Logf("saw patched annotations")
+						sawPatched = true
+					} else if watchedCSC.Annotations["updated"] == "true" {
+						framework.Logf("saw updated annotations")
+						sawUpdated = true
+					} else {
+						framework.Logf("missing expected annotations, waiting: %#v", watchedCSC.Annotations)
+					}
+				} else if evt.Type == watch.Added {
+					_, isCSC := evt.Object.(*storagev1.CSIStorageCapacity)
+					framework.ExpectEqual(isCSC, true, fmt.Sprintf("expected CSC, got %T", evt.Object))
+					sawAdded = true
+				}
+
+			case <-time.After(wait.ForeverTestTimeout):
+				framework.Fail("timed out waiting for watch event")
+			}
+		}
+		cscWatch.Stop()
+
+		// main resource delete operations
+
+		ginkgo.By("deleting")
+		err = cscClient.Delete(context.TODO(), createdCSC.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+		_, err = cscClient.Get(context.TODO(), createdCSC.Name, metav1.GetOptions{})
+		framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %#v", err))
```

Comment:
this will error/flake if a finalizer was added to the object... we shouldn't prevent a cluster from being conformant if they add finalizer behavior on storagecapacities. make sure it either gets a 404 or has a deletionTimestamp set with a finalizer

see the `// If ingress controller does not support finalizers, we expect a 404.  Otherwise we validate finalizer behavior.` bit in the ingress conformance test